Add a custom editable dashboard to observability grafana instance

aabaris opened the pull request.

aabaris committed:
Add a custom editable dashboard to observability grafana instance

Custom dashboard is being created in order to provide an interactively
editable view of observability metrics.

- add configmap containing custom dashboard template
- update nerc-ocp-infra overlay to pull in the new config map

larsks commented on cluster-scope/overlays/nerc-ocp-infra/kustomization.yaml:
```diff
@@ -27,6 +27,7 @@ resources:
 - apiserver/cluster.yaml
 - metrics-bucket-storage
 - redhatcop.redhat.io/odf-node-patcher.yaml
+- configmaps
```
I think I might put this into a "grafana-dashboards" directory rather than a generic "configmap" directory.

aabaris replied:
done

larsks commented on cluster-scope/overlays/nerc-ocp-infra/configmaps/kustomization.yaml:
```diff
@@ -0,0 +1,7 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+commonLabels:
+  nerc.mghpcc.org/kustomized: "true"
```
You don't need to set this here -- it's set by the top level `kustomization.yaml`.

aabaris replied:
also done

larsks reviewed (commented):
I've left a few small comments about these changes.

Generally, I like to see commit messages that include some verbage about *why* we're making the change. [This](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53) is a good concise summary of some best practices.

aabaris committed:
Update custom dashboard files to better match OCP-on-NERC practices

- rename directory to represent custom dashboard rather than configmap
- remove redundant kustomize file header
